homu/server.py

```diff
@@ -29,8 +29,16 @@ def index():
 
 @get('/queue/<repo_label>')
 def queue(repo_label):
-    repo = g.repos[repo_label]
-    pull_states = sorted(g.states[repo_label].values())
+    if repo_label == 'all':
+        labels = list(g.repos.keys())
```

Comment:
That's true, but there's more. The iterator returned by `.keys()` is invalidated when `g.repos` is modified, which is a good thing I think.

For example, if a new repository is added to `g.repos` just after `labels` is retrieved through `.keys()`, the `for` loop below will raise an exception as the iterator is now invalid. However, if `labels` is set using `list(.keys())`, the `for` loop won't fail, which I think is not a good sign.

At the same time, I've realized that, even though Python supports duck typing, it is generally recommended to bind the same type to the same variable name. For instance, the iterator returned by `.keys()` cannot be used with `len()`. So, if someone writes a code that uses `len()` with `labels`, it would work for the most time, but it will occasionally fail if `repo_label` is `'all'`. These kinds of bugs have been hard to find in my experience.

So, there are pros and cons to both with and without `list()`. That's why I cannot decide which to use easily. For now, I'm slightly in favor of not using `list()`, as it makes the iterator invalidation more explicit. However, I also think that may become a source of another bug.

Or, I'm just too obsessive! This isn't really a big deal, to be honest. I'm just poor at making a decision.